swagger.yaml

```diff
@@ -146,226 +328,809 @@ paths:
+      id:
+        description: "An unique id for a dataset edition"
+        readOnly: true
+        type: string
+      links:
+        $ref: '#/definitions/EditionLinks'
+      state:
+        $ref: '#/definitions/State'
   Editions:
     type: object
     properties:
+      count:
+        description: "The number of editions returned for a dataset"
+        readOnly: true
+        type: integer
       items:
         type: array
         items:
           $ref: '#/definitions/Edition'
-  Versions:
+      limit:
+        description: "The number of editions requested for a dataset"
+        type: integer
+      offset:
+        description: "The first row of editions for a dataset to retrieve, starting at 0. Use this parameter as a pagination mechanism along with the limit parameter"
+        type: integer
+      total_count:
+        description: "The total number of editions against a dataset"
+        readOnly: true
+        type: integer
+  EditionLinks:
+    description: "A list of links related to this resource"
+    readOnly: true
+    type: object
+    properties:
+      dataset:
+        $ref: '#/definitions/DatasetLink'
+      self:
+        $ref: '#/definitions/SelfLink'
+      versions:
+        type: object
+        properties:
+          href:
+            description: "A URL to all versions dor an edition of a dataset"
+            example: "http://localhost:8080/datasets/DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC/edition/2017/versions"
+            type: string
+  Event:
+    type: object
+    properties:
+      type:
+        type: string
+        description: |
+          The type of event, this can be;
+          * Info - for an information event
+          * Error - for an error event
+      service:
+        type: string
+        description: "The name of the service which triggered the event"
+      time:
+        type: string
+        description: "The time of the event in ISO 8601 format"
+      message:
+        type: string
+        description: "A description of the event"
+      message_offset:
+        type: string
+        description: "The kafka message offset (Used to locate the kafka message)"
+  Instance:
+    type: object
+    properties:
+      instance_id:
+        type: string
+        readOnly: true
+        description: "A unique id for a job"
+      job:
+        description: "An object describing the ID and URL of the job containing this instance"
+        type: object
+        readOnly: true
+        properties:
+          id:
+            type: string
+            description: "The ID of the job containing this instance"
+            example: 042e216a-7822-4fa0-a3d6-e3f5248ffc35
+          href:
+            type: string
+            description: "The URL for the job containing this instance"
+            example: 'http://localhost:21800/jobs/042e216a-7822-4fa0-a3d6-e3f5248ffc35'
+      state:
+        $ref: '#/definitions/State'
+      events:
```

Comment:
Should this be an array or just a link to the /instances/{id}/events endpoint?

Reply:
Originally it was an array of events, but thinking about it now it should be links to events. Do you want me to changes this now in both the spec and code?